yaormfilter/tools.go

```diff
@@ -153,3 +153,41 @@ func Gte(v interface{}) ValueFilter {
 	}
 	panic(fmt.Errorf("Unknown type: %+v for value %+v in Gte filter", underlyingValue.Kind(), v))
 }
+
+// NotIn returns the correct filter according to the value sent
+func NotIn(values ...interface{}) ValueFilter {
+	var t reflect.Type
+	if tools.GetNonPtrValue(values).Len() == 0 {
+		return nil
+	}
+	for idx, v := range values {
+		underlyingValue := tools.GetNonPtrValue(v)
+		if idx == 0 {
```

Comment:
you can directly declare t as the first array's value's type and then iterate from index 1

Reply:
Is it better now?